source/memory/memory.go

```diff
@@ -179,6 +192,71 @@ func detectNv() ([]nfdv1alpha1.InstanceFeature, error) {
 	return info, nil
 }
 
+// detectHugePages checks whether huge pages are enabled on the node
+// and retrieves the configured huge page sizes.
+func detectHugePages() (map[string]string, error) {
+	hugePages := map[string]string{
+		"enabled": "false",
+	}
+
+	basePath := hostpath.SysfsDir.Path("kernel/mm/hugepages")
+	subdirs, err := os.ReadDir(basePath)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return hugePages, nil
+		}
+		return nil, fmt.Errorf("unable to read huge pages size: %w", err)
+	}
+
+	reg := regexp.MustCompile(`hugepages-(\d+)`)
+	for _, entry := range subdirs {
+		if !entry.IsDir() {
+			continue
+		}
+
+		pageSize, err := getHugePageSize(basePath, entry.Name(), reg)
+		if err != nil {
+			klog.ErrorS(err, "skipping huge page entry", "entry", entry.Name())
+			continue
+		}
+		if pageSize != "" {
+			hugePages["enabled"] = "true"
+			hugePages[pageSize] = "true"
+		}
+	}
+
+	return hugePages, nil
+}
+
+func getHugePageSize(basePath, dirName string, reg *regexp.Regexp) (string, error) {
+	matches := reg.FindStringSubmatch(dirName)
+	if len(matches) != 2 {
+		return "", fmt.Errorf("unexpected directory format: %s", dirName)
+	}
+
+	totalPagesFile := filepath.Join(basePath, dirName, "nr_hugepages")
+	totalPagesRaw, err := os.ReadFile(totalPagesFile)
+	if err != nil {
+		return "", fmt.Errorf("unable to read total number of huge pages from the file: %s", totalPagesFile)
+	}
+
+	totalPages, err := strconv.Atoi(strings.TrimSpace(string(totalPagesRaw)))
+	if err != nil {
+		return "", fmt.Errorf("unable to convert total pages to integer - %s: %s", totalPagesFile, totalPagesRaw)
+	}
+	if totalPages < 1 {
+		return "", nil
+	}
+
+	sizeFormat := matches[1] + "Ki"
+	quantity, err := resource.ParseQuantity(sizeFormat)
+	if err != nil {
+		return "", fmt.Errorf("invalid size format: %s", sizeFormat)
+	}
+
+	return "hugepages-" + quantity.String(), nil
```

Comment:
If we want to indicate that the naming matches with Kubernetes resource names we could possibly use `corev1.ResourceHugePagesPrefix` here.

Reply:
Done.